src/h2olog/misc/gen_raw_tracer.py

```diff
@@ -683,16 +682,18 @@ def generate_cplusplus(context, output_file):
 
   handle_event_func = r"""
 void h2o_raw_tracer::do_handle_event(const void *data, int data_len) {
-  const h2olog_event_t *event = static_cast<const h2olog_event_t*>(data);
+  // The perf event data is not aligned, so we use a local copy to avoid UBSan errors.
+  // cf. https://github.com/iovisor/bpftrace/pull/1520
+  const h2olog_event_t event = *static_cast<const h2olog_event_t*>(data);
```

Comment:
I'm not sure if this change eliminates unaligned access.

I think that the fact `event->id` was not properly aligned means that at least one property of `static_cast<const h2olog_event_t *>(data)` was misaligned.

Therefore, the correct fix here would be to use `memcpy` to copy the data, as memcpy is defined as a function being capable of copying "bytes."

Reply:
Nice catch.

What you mean is that copying a struct with `=` logically equals to copying each property one by one, so if a property is misaligned, `=` could lead to a UB as well as the use of the misaligned property could lead, right?

Fixed d1b7028e546439f6dc647dbef220a66276d96460

Reply:
Correct.

Or to paraphrase, C mandates that the alignment of a struct should be determined so that none of its members would be misaligned. The fact that a misalignment was detected for one member means that `data` is misaligned when accessed as `h2o_log_event_t *`.

Reply:
I see. Thanks for clarifying it.